contracts/dapis/DapiServer.sol

```diff
@@ -699,13 +699,31 @@ contract DapiServer is
         return (dataFeed.value, dataFeed.timestamp);
     }
 
-    /// @notice Reads the dAPI with name
+    /// @notice Reads the data feed value with ID
+    /// @param dataFeedId Data feed ID
+    /// @return value Data feed value
+    function readDataFeedValueWithId(bytes32 dataFeedId)
+        external
+        view
+        override
+        returns (int256 value)
+    {
+        require(
+            readerCanReadDataFeed(dataFeedId, msg.sender),
+            "Sender cannot read"
+        );
+        DataFeed storage dataFeed = dataFeeds[dataFeedId];
+        require(dataFeed.timestamp != 0, "Data feed does not exist");
```

Comment:
Do you mind sharing why we now need to validate that the dataFeed exist while reading?

Reply:
Normally the reader gets a struct consisting of the value and the timestamp, and is responsible with validating the timestamp (to check if it's fresh enough, etc) Here we're discarding the timestamp, which disables the user from doing that. The least that we can do is to check if the timestamp is 0 because you probably never want to use the value of an uninitialized data feed (which will be 0) in any kind of logic.

There's also an argument to be made towards "you should never half-validate to avoid creating a sense of false security" so this is somewhat subjective. I think the general approach in contract development is to make a best effort in validation though and this is consistent with that.